### PR TITLE
fix respond on SDL.GetListOfPermission request

### LINKED ISSUE #3556: SDL does not respond to SDL.GetListOfPermission with appId in case there are no specific permissions for the app
### Bug Report
SDL does not respond to `SDL.GetListOfPermission` with appId in case there are no specific permissions for the app

##### Reproduction Steps
1. SDL with external_proprietary policies and HMI are started
2. Mobile app is registered and activated
3. There are no specific permissions for the app
4. PTU is not performed or performed without specific permissions for the mobile app
5. Go to HMI settings-> App permissions - > \<app name\>
6. HMI requests `{"jsonrpc":"2.0","id":<n>,"method":"SDL.GetListOfPermissions","params":{"appID":<app_id>}}`

##### Expected Behavior
SDL responds with `{"id":<n>,"jsonrpc":"2.0","result":{"allowedFunctions":[],"code":0,"externalConsentStatus":[],"method":"SDL.GetListOfPermissions"}}`

##### Observed Behavior
SDL does not respond to `SDL.GetListOfPermissions` request

##### OS & Version Information
* OS/Version: Ubuntu 18.04
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/commit/54246454776513b68161258b46299b48e4352f3f
* Testing Against: HMI: https://github.com/smartdevicelink/sdl_hmi/commit/931877cb68e56522d9f6db1d6c5751decf6c988e, SPT: 20201007

##### SDL log
[SDL_log_no_response_to_GetListOfPermissions.zip](https://github.com/smartdevicelink/sdl_core/files/5457962/SDL_log_no_response_to_GetListOfPermissions.zip)

### PR DESCRIPTION
**Pull request by VadymLuchko**
Fixes #3556

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Covered by unit test

### Summary
There was noticed an incorrect behavior in PolicyHandler::OnGetListOfPermissions method.
No response was sent if the list of permission was empty.
In this fix we check if we went through all the methods for filling the List Of Permissions and didn't cause an error, then we send an empty list.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by VadymLuchko** [comment]
@JackLivio  this PR is ready for review

File: src/components/application_manager/include/application_manager/policies/policy_handler.h [hunk 1]
```diff
@@ -876,19 +876,21 @@ class PolicyHandler : public PolicyHandlerInterface,
...
    * @brief Collects permissions for all currently registered applications on
    * all devices
-   * @return consolidated permissions list or empty list if no
-   * applications/devices currently present
+   * @param outPermissions output ref on list of application permissions
+   * @return true if app connection information is valid, otherwise - false
    */
-  std::vector<FunctionalGroupPermission> CollectRegisteredAppsPermissions();
+  bool CollectRegisteredAppsPermissions(
+      std::vector<FunctionalGroupPermission>& outPermissions);
```
Comment (jacobkeeler) [inline src/components/application_manager/include/application_manager/policies/policy_handler.h:883]:
```suggestion
      std::vector<FunctionalGroupPermission>& out_permissions);
```

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 2]
```diff
@@ -935,23 +936,26 @@ void PolicyHandler::OnGetListOfPermissions(const uint32_t connection_key,
...
+        policy_manager->GetExternalConsentStatus(),
 #endif  // EXTERNAL_PROPRIETARY_MODE
-      correlation_id,
-      application_manager_);
+        correlation_id,
+        application_manager_);
+  } else {
+    SDL_LOG_ERROR(
+        "Permissions collection failed for application with connection key:"
+        << connection_key);
```
Comment (jacobkeeler) [inline src/components/application_manager/src/policies/policy_handler.cc:957]: Should we not send an error response in this case?

**Review by jacobkeeler — CHANGES_REQUESTED** [review]

File: src/components/application_manager/src/policies/policy_handler.cc (hunk 2 above)
Comment (VadymLuchko) [reply src/components/application_manager/src/policies/policy_handler.cc:958]: @jacobkeeler done in  01c1aa6dc2c32f07117516a982589085c16906df
I made some refactor of this scope and now we send GetListOfPermission response to HMI anyway, but with error if collection result is false;

**Review by jacobkeeler — CHANGES_REQUESTED** [review]

File: src/components/application_manager/test/include/application_manager/mock_message_helper.h [hunk 3]
```diff
@@ -139,18 +139,20 @@ class MockMessageHelper {
...
+           const bool success_flag));
 #else
-  MOCK_METHOD3(
+  MOCK_METHOD4(
       SendGetListOfPermissionsResponse,
       void(const std::vector<policy::FunctionalGroupPermission>& permissions,
            uint32_t correlation_id,
-           ApplicationManager& app_mngr));
+           ApplicationManager& app_mngr,
+           const bool sucess_flag));
```
Comment (jacobkeeler) [inline src/components/application_manager/test/include/application_manager/mock_message_helper.h:155]:
```suggestion
           const bool success_flag));
```

**Comment by jacobkeeler** [comment]
@VadymLuchko one more missed rename

File: src/components/application_manager/test/include/application_manager/mock_message_helper.h (hunk 3 above)
Comment (VadymLuchko) [reply src/components/application_manager/test/include/application_manager/mock_message_helper.h]: Thanks, I didn't notice it

**Review by jacobkeeler — APPROVED** [review]

**Pull request merged**